apps/accelerate/open_alpha_tensor/open_alpha_tensor/core/actors/stage.py

```diff
@@ -34,24 +44,25 @@ def remove_duplicates(reducing_tensor: torch.Tensor):
                 idx_map[idx] = []
 
     # idx_map = {i: len(v) for i, v in enumerate(idx_map.values())}
-    old_idx_to_new_idx_map = {}
-    for new_idx, (key, values) in enumerate(idx_map.items()):
-        old_idx_to_new_idx_map[key] = new_idx
+    cloned_idx_to_idx_map = {}
+    for key, values in idx_map.items():
         for second_idx in values:
-            old_idx_to_new_idx_map[second_idx] = new_idx
-    return (
-        reducing_tensor[:, indexes],
-        old_idx_to_new_idx_map,
-        idx_map,
-        indexes,
-    )
+            cloned_idx_to_idx_map[second_idx] = key
+    return reducing_tensor[:, indexes], cloned_idx_to_idx_map, indexes
```

Comment:
please keep the old code modification (the code in red should replace the code in green)